lib/mayaUsd/ufe/UsdContextOps.cpp

```diff
@@ -567,15 +569,10 @@ Ufe::UndoableCommand::Ptr UsdContextOps::doOpCmd(const ItemPath& itemPath)
         return std::make_shared<SetVariantSelectionUndoableCommand>(prim(), itemPath);
     } // Variant sets
     else if (itemPath[0] == kUSDToggleVisibilityItem) {
-        auto attributes = Ufe::Attributes::attributes(sceneItem());
-        assert(attributes);
-        auto visibility = std::dynamic_pointer_cast<Ufe::AttributeEnumString>(
-            attributes->attribute(UsdGeomTokens->visibility));
-        assert(visibility);
-        auto current = visibility->get();
-        return visibility->setCmd(
-            current == UsdGeomTokens->invisible ? UsdGeomTokens->inherited
-                                                : UsdGeomTokens->invisible);
+        auto object3d = UsdObject3d::create(fItem);
+        auto current = object3d->visibility();
+        auto cmd = object3d->setVisibleCmd(!current);
+        cmd->execute();
```

Comment:
Doesn't this new code need to be protected by UFE_PREVIEW_VERSION_NUM >= 2034 and leave the old code for the else?

Reply:
Ugh.... Good catch! Indeed. :)

Reply:
Addressed in 70046317e72410a8a661614b42f579146509cba3